src/share/poudriere/common.sh

```diff
@@ -1427,14 +1427,17 @@ get_data_dir() {
 			zfs get -H -o value mountpoint ${ZPOOL}${ZROOTFS}/data
 			return
 		fi
-		zfs create -p -o ${NS}:type=data \
-			-o atime=off \
-			-o mountpoint=${BASEFS}/data \
-			${ZPOOL}${ZROOTFS}/data
+		# Set properties on top dataset and let underlying ones inherit them
+		# Explicitly set properties for values diverging from top dataset
+		zfs create -p -o atime=off \
+			-o compression=lz4 \ 
```

Comment:
I think this should just be `compression=on`.

>       Setting compression to on indicates that the current default compression algorithm should be used.  The default balances compression and decompression speed, with compression ratio and is expected to work well on a wide variety of workloads.  Unlike all other settings for this property, on does not select a
>       fixed compression type.  As new compression algorithms are added to ZFS and enabled on a pool, the default compression algorithm may change.  The current default compression algorithm is either lzjb or, if the lz4_compress feature is enabled, lz4.

> if the lz4_compress feature is enabled, lz4.

That is also more in spirit of #794.